### ISSUE TITLE
[Feature Request] Force apk downloads for a particular architecture

### ISSUE THREAD

**Issue by late-latte** [yeriomin/YalpStore#477]
I downloaded Chrome some time ago, but what I got apparently is the armv7 (32 bit)version. While it's operational on its own, it caused Firefox Klar to crash because it wanted armv8 (64 bit) version, and my phone itself is an aarch64. No matter which phone I masqueraded as, it still downloaded the 32 bit one (I've tried pixel 2, oneplus 3t, pretty much every phone model I know is using aarch64). I've even used my model to no avail. Considering that aarch64 would benefit greatly by using 64 bit version of apps in general as well, is it possible that you make this app to force download 64 bit? And maybe a check system to determine which version is being downloaded?

**Comment by yeriomin** [comment]
> I downloaded Chrome some time ago, but what I got apparently is the armv7 (32 bit)version.
> No matter which phone I masqueraded as, it still downloaded the 32 bit one

Are you sure?  You can check by opening the apk file as archive and looking into "lib" directory. It had "armeabi-v7a" and "arm64-v8a" subdirectories in the chrome apk I downloaded. Even if you got the armv7 version, there might be only one version in Play Store for each app, this depends on each app's developer.

> it caused Firefox Klar to crash because it wanted armv8 (64 bit) version

Again, how do you know that? What was the message? It seems strange to me that one browser depends on another. Maybe you meant webview?

> is it possible that you make this app to force download 64 bit?

No. The decision about what apk is available is done on Google's side. Your device's supported architectures are sent during login (and have always been since version 0.1).

**Comment by switchtegrax1** [comment]
if i recall correctely standard chrome is only 32bits,you need chrome beta ou canary to get 64bits

**Comment by late-latte** [comment]
I did download both Chrome and Android System Webview/Chromium to make use either webview (as you most likely know you can choose implementation in Developer Options). I even made sure I checked and try using both 32 and 64 bit, and all 4 of them actually throw errors, albeit with different error statements between 32 and 64 bit ones. This happened with at least Firefox Klar and KeePassDroid (I just knew KeePassDroid is using webview! But there are some apps using webview (i.e. Riot) that are running fine, so...). I'm starting to think that it's possibly a bad implementation by the OEM (as I know LG did hesitate in releasing Nougat for LG V10, and it's apparently beta-ish 7.0 release from skimming  their source code, as you can see from: http://opensource.lge.com/ <http://opensource.lge.com/>).

Anyway, tldr: the issue isn't with Yalp Store. Keep up the good work and sorry for the distraction.

Jun 17, 2018, 11:12 PM by notifications@github.com:
